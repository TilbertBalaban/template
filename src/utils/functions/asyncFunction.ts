import axios from 'axios';
import { STAR_WARS_API_URL } from '../api/consts';

export const asyncFunction = async () => {
  try {
    const { data } = await axios.get(`${STAR_WARS_API_URL}vehicles/`);
    return data.count;
  } catch (e) {
    return e;
  }
};
