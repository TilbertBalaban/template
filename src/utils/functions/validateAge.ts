export const validateAge = (age: number): string => {
  if (age < 0 || age > 150 || Number.isNaN(age)) {
    return 'Invalid';
  }
  return age < 18 ? 'You are too young' : 'You are an adult';
};
