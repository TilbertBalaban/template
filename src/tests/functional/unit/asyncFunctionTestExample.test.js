import axios from 'axios';
import { asyncFunction } from '../../../utils/functions/asyncFunction';

jest.mock('axios');

describe('asyncFunction', () => {
  let response;
  beforeEach(() => {
    response = {
      data: {
        count: 40,
      },
    };
  });
  test('async', async () => {
    axios.get.mockReturnValue(response);
    const result = await asyncFunction();
    expect(axios.get).toBeCalledTimes(1);
    expect(result).toBe(40);
    expect(result).toMatchSnapshot();
  });
});
