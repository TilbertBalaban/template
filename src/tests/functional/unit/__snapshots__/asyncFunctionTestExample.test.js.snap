// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asyncFunction async 1`] = `40`;
